misc/futures-bounded/src/list.rs

```diff
@@ -0,0 +1,58 @@
+use std::future::Future;
+use std::task::{ready, Context, Poll};
+use std::time::Duration;
+
+use futures_util::future::BoxFuture;
+
+use crate::{FuturesMap, PushError, Timeout};
+
+/// Represents a list of [Future]s.
+///
+/// Each future must finish within the specified time and the list never outgrows its capacity.
+pub struct FuturesList<O> {
+    id: i32,
+    inner: FuturesMap<i32, O>,
+}
+
+impl<O> FuturesList<O> {
+    pub fn new(timeout: Duration, capacity: usize) -> Self {
+        Self {
+            id: i32::MIN,
+            inner: FuturesMap::new(timeout, capacity),
+        }
+    }
+}
+
+impl<O> FuturesList<O> {
+    /// Push a future into the list.
+    ///
+    /// This method adds the given future to the list.
+    /// If the length of the list is equal to the capacity, this method returns a error that contains the passed future.
+    /// In that case, the future is not added to the set.
+    pub fn try_push<F>(&mut self, future: F) -> Result<(), BoxFuture<O>>
+    where
+        F: Future<Output = O> + Send + 'static,
+    {
+        (self.id, _) = self.id.overflowing_add(1);
```

Comment:
```suggestion
        (self.id, _) = self.id.checked_add(1).expect("i32 to never be exhausted");
```

We expect this to never overflow, correct? Overflowing would panic further below, correct? If so, let's panic here already.

Reply:
No, 'id' can overflow. 

> Overflowing would panic further below, correct?

No, we get a negative value for `id`.
We just need a unique `id` here, so if it would be negative it's ok.

Reply:
But I can make this a simpler way.